uint8 direction
uint32 vel
int32 distance
int32 maxTime
